lib/ansible/playbook/role/requirement.py

```diff
@@ -61,8 +61,12 @@ def __init__(self):
     def repo_url_to_role_name(repo_url):
         # gets the role name out of a repo like
         # http://git.example.com/repos/repo.git" => "repo"
+        # "/tmp/hello-test.tar.gz"                => "hello-test"
 
         if '://' not in repo_url and '@' not in repo_url:
+            trailing_path = repo_url.split('/')[-1]
```

Comment:
You're right, of course.

Only thing is, I copied those lines... They are used in the same function.